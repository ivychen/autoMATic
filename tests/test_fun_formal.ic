/* autoMATic Compiler - Spring 2018 - COMS4115 PLT
 by Ivy Chen ic2389, Nguyen Chi Dung ncd2118,
 Nelson Gomez ng2573, Jimmy O'Donnell jo2474 */

int main() {
    int x = 4;
    string y = "test";

    test1(x, "test");
    return 0;
}

// main declarations of "x" and "y" should not conflict
void test1(int x, string y) {
    println(x);
    println(y);
    test2(x);
}

// test2's formal "x" should not conflict with test1's
void test2(int x) {
    println(x);
}
