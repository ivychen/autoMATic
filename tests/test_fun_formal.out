4
test
4
